analytics-core/src/main/java/com/liferay/analytics/android/Analytics.kt

```diff
@@ -59,8 +62,19 @@ class Analytics private constructor(
 				throw IllegalArgumentException("AppContext can't be null.")
 			}
 
-			if (analyticsKey.isNullOrEmpty() || analyticsKey.isBlank()) {
-				throw IllegalArgumentException("Analytics key can't be null or empty.")
+			val applicationInfo =
+				context.packageManager.getApplicationInfo(context.packageName, PackageManager.GET_META_DATA)
+
+            val dataSourceId = applicationInfo.metaData.getString("com.liferay.analytics.DataSourceId")
+
+			if (dataSourceId.isNullOrEmpty() || dataSourceId.isBlank()) {
+				throw IllegalArgumentException("Analytics Data source id can't be null or empty.")
+			}
+
+			val endpointURL = applicationInfo.metaData.getString("com.liferay.analytics.EndpointUrl")
+
+			if (endpointURL.isNullOrEmpty() || endpointURL.isBlank() || HttpUrl.parse(endpointURL) == null) {
+				throw IllegalArgumentException("You must provide a valid endpoint URL")
```

Comment:
Maybe we should extract these validation checks to another function, this function is getting a bit longer than the expected.